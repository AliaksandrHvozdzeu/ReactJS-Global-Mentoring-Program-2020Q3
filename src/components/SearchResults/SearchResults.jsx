import React from 'react';
import PropTypes from 'prop-types';
import NotFound from '../NotFoundMovie';
import './style.css';

export default function SearchResults({ searchResults }) {

  const MovieResults = React.lazy(() => import('../MovieResultList'));

  return (
    <>
      {searchResults && (
        <React.Suspense fallback={<h3>Loading... Please wait.</h3>}>
          <MovieResults results={searchResults}/>
        </React.Suspense>
      )}
      {!searchResults && <NotFound/>}
    </>
  );

}

SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      poster: PropTypes.string.isRequired,
      genres: PropTypes.array.isRequired,
      release: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
    }).isRequired,
  ).isRequired,
};
