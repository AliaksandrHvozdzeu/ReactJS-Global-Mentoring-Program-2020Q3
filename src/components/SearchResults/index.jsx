import React  from 'react';
import PropTypes from 'prop-types';
import MovieResults from '../MovieResultList';
import NotFound from '../NotFoundMovie';
import './style.css';

const SearchResults = ({ searchResults }) => (
  <>
    {searchResults && <MovieResults results={searchResults} />}
    {!searchResults && <NotFound />}
  </>
);

SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      genre: PropTypes.string.isRequired,
      release: PropTypes.string.isRequired,
      rating: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
};

export default SearchResults;
