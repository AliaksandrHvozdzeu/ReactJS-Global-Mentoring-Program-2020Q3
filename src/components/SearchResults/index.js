import SearchResults from './js/SearchResults';

export default SearchResults;
