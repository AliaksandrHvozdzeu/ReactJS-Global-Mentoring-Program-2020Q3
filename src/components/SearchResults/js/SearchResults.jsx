import React from 'react';
import '../css/SearchResults.css';
import MovieLoadMessage from '../../MovieLoadMessage';

export default function SearchResults() {

  const MovieResults = React.lazy(() => import('../../MovieResultList'));

  const loadWindow = (
    <MovieLoadMessage/>
  );

  return (
    <>
      <React.Suspense fallback={loadWindow}>
        <MovieResults/>
      </React.Suspense>
    </>
  );

}
