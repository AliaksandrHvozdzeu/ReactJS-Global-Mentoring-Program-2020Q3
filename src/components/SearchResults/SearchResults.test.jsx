import React from 'react';
import rerender from 'react-test-renderer';
import SearchResults from './js/SearchResults';

describe('when SearchResults', () => {

  test('then snapshot created', () => {
    const component = rerender.create(<SearchResults/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
