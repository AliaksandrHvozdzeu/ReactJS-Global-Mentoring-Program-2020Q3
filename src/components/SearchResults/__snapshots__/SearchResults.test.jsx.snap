// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when SearchResults then snapshot created 1`] = `
Array [
  <div
    className="modal-window-overlay-message"
  />,
  <div
    className="modal-window"
  >
    <div
      className="modal-window-loader"
    >
      Loading... Please wait!
    </div>
  </div>,
]
`;
